core/listings.go

```diff
@@ -1175,6 +1175,27 @@ func validateListing(listing *pb.Listing, testnet bool) (err error) {
 	return nil
 }
 
+func ValidShippingRegion(shippingOption *pb.Listing_ShippingOption) int {
+	for _, region := range shippingOption.Regions {
+		if int32(region) == 0 {
+			return 1
+		}
+		_, ok := proto.EnumValueMap("CountryCode")[region.String()]
+		// if region not in country codes return 2
+		if !ok {
+			// not in CountryCodes
+			return 2
+		}
+		if ok {
+			// region exists but it is a continent
+			if int32(region) > 500 {
+				return 3
+			}
+		}
+	}
+	return 0
+}
```

Comment:
Please return a descriptive `error` instead of an error number? If you like, you can enumerate them like so:
```
var (
  ErrShippingRegionMustBeSet = errors.New("shipping region must be set")
  ErrShippingRegionUndefined = errors.New("undefined shipping region")
  ErrShippingRegionMustNotBeContinent = errors.New("cannot specify continent as shipping region")
)

func ValidShippingRegion(option) error {
  ...
  if int32(region) > 500 {
    return ErrShippingRegionMustNotBeContinent
  }
  ...
  return nil
}

if err := ValidShippingRegion(shippingOption); err != nil {
  return fmt.Errorf("invalid shipping option (%s): %s", shippingOption.String(), err.Error())
}
```